### DISCUSSION TITLE
Процессор от роутера Asus RT-AC87U не поддерживается

### DISCUSSION THREAD

**Discussion by ToucanB67** [bol-van/zapret#196] [General]
Столкнулся с проблемой, хотел поставить программку на asuswrt merlin и dd-wrt но произошла проблема по бинарникам… как я понял процессор от rt ac87u не поддерживается, он на архитектуре armv7-a Broadcom BCM4709A (ARM Cortex-A9)

Как вообще эту программу скомпилировать?

**Comment by bol-van** [comment]
У меня собрана версия под bcm2708 (raspbery pi 1) с soft float.  Даже не armv7, а  armv6 (ARM11).
Работает даже на древнем телике samsung 2012 года.

Что происходит при запуске exe-шников из binaries/arm ?
Если их распаковать upx и запустить что выходит ?

**Reply by ToucanB67** [reply]
Я попробовал напрямую запустить exec файл и курсор перемешается в низ и ничего не происходит просто виснет. Распаковал ip2net через upx (получил просто файл в два раза больше чем изначально) отправил его на роутер и результат тот же

**Comment by bol-van** [comment]
ip2net ничего и не должен выводить
он ждет в stdin список ip адресов

правильно так : echo 1.1.1.1 | ./ip2net

но судя по всему оно работает. если бы там была какая-то несовместимая инструкция процессора - оно бы вывалилось по эксепшину. с чего сделан вывод о неработе ?
install_bin.sh что выводит ?

**Reply by ToucanB67** [reply]
Увы но install_bin.sh выводит все бинарники is not ok в независимости от прошивки

**Reply by bol-van** [reply]
> Увы но install_bin.sh выводит все бинарники is not ok в независимости от прошивки

может права на EXEшники слетели ?
что происходит при запуске, допустим, binaries/arm/tpws -h

**Reply by mad0k** [reply]
Была такая же ситуация. В install_bin.sh в кейсе проверки архитектуры добавил armv71 в Linux, в binaries скопировал директорию arm, переименовав ее в armv71. Поначалу скрипт по какой-то причине не отрабатывал, но после ребута роутера инсталляция прошла успешно. Пользуюсь tpws + hostfile с определенными доменами.

П.С.: Не знаю насколько верно указал архитектуру, но lscpu показывала именно эту.

**Reply by ToucanB67** [reply]
> Была такая же ситуация. В install_bin.sh в кейсе проверки архитектуры добавил armv71 в Linux, в binaries скопировал директорию arm, переименовав ее в armv71. Поначалу скрипт по какой-то причине не отрабатывал, но после ребута роутера инсталляция прошла успешно. Пользуюсь tpws + hostfile с определенными доменами.
> 
> П.С.: Не знаю насколько верно указал архитектуру, но lscpu показывала именно эту.

Надо будет попробовать)

**Reply by ToucanB67** [reply]
> > Увы но install_bin.sh выводит все бинарники is not ok в независимости от прошивки
> 
> может права на EXEшники слетели ? что происходит при запуске, допустим, binaries/arm/tpws -h

оно выдает оч много текста где очень много команд, как я понял оно в целом работает но как починить arm is not ok я не знаю....

**Reply by ToucanB67** [reply]
> Была такая же ситуация. В install_bin.sh в кейсе проверки архитектуры добавил armv71 в Linux, в binaries скопировал директорию arm, переименовав ее в armv71. Поначалу скрипт по какой-то причине не отрабатывал, но после ребута роутера инсталляция прошла успешно. Пользуюсь tpws + hostfile с определенными доменами.
> 
> П.С.: Не знаю насколько верно указал архитектуру, но lscpu показывала именно эту.

попробовал, оно пишет тоде самое только прибавилось armv71 is not ok

**Reply by bol-van** [reply]
значит в вашей системе нет команды  find
или урезаная версияот бизибокс без -exec
find это единственный способ в openwrt выполнить exec, чтобы шелл не стал интерпоетировать бинарь как текстовый скрипт
переименовывать директории смысла нет
архитектура определяется по возможности выполнить бинарь

обшее заключение прошивка не поддерживается
выход править код инсталл бин чтобы он всегда успешно выбирал  arm

**Reply by ToucanB67** [reply]
> значит в вашей системе нет команды find или урезаная версияот бизибокс без -exec find это единственный способ в openwrt выполнить exec, чтобы шелл не стал интерпоетировать бинарь как текстовый скрипт переименовывать директории смысла нет архитектура определяется по возможности выполнить бинарь
> 
> обшее заключение прошивка не поддерживается выход править код инсталл бин чтобы он всегда успешно выбирал arm

вообщем я уже решил конкретную проблему, я попросил чат гпт переделать install bin так чтобы убрать полностью проверку и выбирало принудительно папку с арм) и это сработало)

**Reply by bol-van** [reply]
оригинально. скоро железки за нас проги писать начнут. и тогда нам трындец

**Comment by sad-spirit** [comment]
Напоролся на ту же самую проблему (dd-wrt, ASUS'овский роутер на ARM). Там на самом деле find из busybox не понимает опцию `-maxdepth`:
```
BusyBox v1.37.0 (2024-12-05 04:46:15 +07) multi-call binary.

Usage: find [-HL] [PATH]... [OPTIONS] [ACTIONS]

Search for files and perform actions on them.
First failed action stops processing of current file.
Defaults: PATH is current directory, action is '-print'

        -L,-follow      Follow symlinks
        -H              ...on command line only
        -xdev           Don't descend directories on other filesystems

Actions:
        ACT1 [-a] ACT2  If ACT1 fails, stop, else do ACT2
        ACT1 -o ACT2    If ACT1 succeeds, stop, else do ACT2
                        Note: -a has higher priority than -o
        -name PATTERN   Match file name (w/o directory name) to PATTERN
        -iname PATTERN  Case insensitive -name
        -type X         File type is X (one of: f,d,l,b,c,s,p)
        -perm MASK      At least one mask bit (+MASK), all bits (-MASK),
                        or exactly MASK bits are set in file's mode
        -mtime DAYS     mtime is greater than (+N), less than (-N),
                        or exactly N days in the past
        -newer FILE     mtime is more recent than FILE's
        -inum N         File has inode number N
If none of the following actions is specified, -print is assumed
        -print          Print file name
        -print0         Print file name, NUL terminated
        -exec CMD ARG ; Run CMD with all instances of {} replaced by
                        file name. Fails if CMD exits with nonzero
        -execdir CMD ARG ;Run CMD with all instances of {} replaced by
                        file name from its path. Fails if CMD exits with nonzero

```

Если её убрать у вызова в `install_bin.sh`, то архитектура определяется правильно.

**Reply by bol-van** [reply]
Прошивки, кроме openwrt, официально не поддерживаются. Их любят кастрировать различным образом, исключая те или иные команды или их параметры. Угнаться за всем очень сложно
